rococo/data/mongodb.py

```diff
@@ -0,0 +1,71 @@
+from typing import Any, Dict, List, Tuple, Union
+from rococo.data.base import DbAdapter
+from pymongo import MongoClient
+
+
+class MongoDBAdapter(DbAdapter):
+    """MongoDB adapter for interacting with MongoDB."""
+
+    def __init__(self, mongo_uri: str, mongo_database: str):
+        self.client = MongoClient(mongo_uri)
+        self.db_name = mongo_database
+        self.db = self.client.get_database(mongo_database)
```

Comment:
If we are adding code dealing with the resource cleanup on `__exit__`,  I'd expect to have resource acquisition in `__enter__`, rather than in `__init__`